Source/WebCore/loader/PolicyChecker.cpp

```diff
@@ -284,7 +284,10 @@ void PolicyChecker::checkNavigationPolicy(ResourceRequest&& request, const Resou
         return;
     }
 
-    auto documentLoader = frameLoader->loaderForWebsitePolicies();
+    // PolicyDecisionMode::Synchronous means that it is a FragmentNavigation and in that case we should use documentLoader,
+    // because there can be ongoing (in policy or provisional state) navigation.
+    const auto canIncludeCurrentDocumentLoader = policyDecisionMode == PolicyDecisionMode::Synchronous ? FrameLoader::CanIncludeCurrentDocumentLoader::Always:FrameLoader::CanIncludeCurrentDocumentLoader::Yes;
```

Comment:
This seems a little hacky, indirectly making an inference.  Are we sure that there aren't fragment navigations with asynchronous policy decisions?

Reply:
`policyChecker().checkNavigationPolicy` with `PolicyDecisionMode::Synchronous` is called in two places and each of them are inside `if (shouldPerformFragmentNavigation(isFormSubmission, httpMethod, newLoadType, newURL))`, so that is why I add this comment.